[ESP32] drivers: Add Entropy Device tree support

ExtremeGTX opened the pull request:
- Add TRNG module to esp32.dtsi
- Updated the note about TRNG register address

Related issues:
- #10516 

Signed-off-by: Mohamed ElShahawi <ExtremeGTX@hotmail.com>

ulfalizer commented on dts/bindings/rng/espressif,esp32-trng.yaml:
```diff
@@ -0,0 +1,21 @@
+title: Espressif ESP32 TRNG (True Random Number Generator)
+
+description: >
+    This binding gives a base representation of the Espressif ESP32 TRNG
```
Know you're just being consistent, but maybe this could give some more information instead, or be shortened to the same thing as the title.

"X" and "This binding gives a base representation of X" are really saying exactly the same thing, just with more words for the second one. Feels like the purpose of the description text got lost somewhere. :)

ExtremeGTX replied:
Is it better to include something about TRNG it self, like how it get the random value or something ?

any suggestions are welcome.

ulfalizer replied:
Yeah, think that might be helpful.

Could go for something like

```
description: Espressif ESP32 TRNG. Gets random values from ...
```

Or whatever other information might be useful to people.

ulfalizer replied:
The description shows up as a comment above the generated #defines in the header by the way.

ExtremeGTX replied:
I changed it to :
```
The TRNG use the noise in the Wi-Fi/BT RF system.
    When Wi-Fi and BT are disabled, the random number generator will give out
    pseudo-random numbers.
``

ulfalizer replied:
Looks good. Thanks!

Could maybe mention the name of the device too, so that it shows up in the comment, but not obligatory.

```
Espressif ESP32 TRNG. Uses the noise in the Wi-Fi/BT RF system.
When Wi-Fi and BT are disabled, the random number generator will give out
pseudo-random numbers.
```

ExtremeGTX committed:
esp32: drivers: Add Entropy Device tree support

- Add TRNG module to esp32.dtsi
- Updated the note about TRNG register address

Signed-off-by: Mohamed ElShahawi <ExtremeGTX@hotmail.com>

ExtremeGTX commented:
should i enable TRNG by default ?
`status = "okay" `
